pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -462,6 +621,139 @@ pub mod pallet {
 
 			Ok((address, incoming_msg))
 		}
+
+		/// Iterates over the outbound messages stored in the queue and attempts
+		/// to process them.
+		///
+		/// If a message fails to process it is moved to the
+		/// `FailedOutboundMessages` storage so that it can be executed again
+		/// via the `process_failed_outbound_message` extrinsic.
+		fn service_outbound_message_queue(max_weight: Weight) -> Weight {
+			let mut weight_used = Weight::zero();
+
+			let mut processed_entries = Vec::new();
+
+			for (nonce, (domain, sender, message)) in OutboundMessageQueue::<T>::iter() {
+				processed_entries.push(nonce);
+
+				let mut weight = Weight::zero();
+
+				match Self::process_message(domain.clone(), sender.clone(), message.clone()) {
+					Ok(post_info) => {
+						Self::deposit_event(Event::OutboundMessageExecutionSuccess {
+							nonce,
+							sender,
+							domain,
+							message,
+						});
+
+						weight = weight
+							.saturating_add(T::DbWeight::get().writes(1))
+							.saturating_add(
+								post_info
+									.actual_weight
+									.expect("expected message processing weight"),
+							);
```

Comment:
Two nits:
1. Cleaner way is to use `weight.saturating_accrue` 
2. The expect message is a bit verbose. Since expecting can panic and thus should generally not be used, I'd prefer to point to the `Ok(...)` match here either via comment or the expect message itself to make it explicit.
```suggestion
						weight.saturating_accrue(T::DbWeight::get().writes(1)
							.saturating_accrue(
								post_info
									.actual_weight
									.expect("Message processing success already ensured; qed"),
							);
```

Reply:
`saturating_accrue` cannot be used like that since it's return is `()`. I suggest we keep the `add` as it is since it's easier to read imo.

Reply:
Whatever you prefer. Of course we don't reassign when using `accrue` but just alter the mutable `weight` variable.